### DISCUSSION TITLE
Modify data representation and field order of response model

### DISCUSSION THREAD

**Discussion by jd-solanki** [fastapi/fastapi#8823] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# TIP: This is full app. Run using `uvicorn main:app --port 5000 --reload`

from datetime import datetime
from typing import List

from fastapi import FastAPI
from pydantic import BaseModel

# Fake Data (database)

data = [
    {
        "id": 1,
        "username": "johndoe007",
        "created_at": datetime.now(),
        "updated_at": datetime.now(),
    }
]


# Pydantic
class DemoBase(BaseModel):
    username: str


class DemoOut(DemoBase):
    id: int
    created_at: datetime
    updated_at: datetime


# FastAPI
app = FastAPI()


@app.get("/", response_model=List[DemoOut])
def read_root():
    return data
```

### Description

If you visit the app you will get below response:

```json
[
    {
        "username": "johndoe007",
        "id": 1,
        "created_at": "2021-09-13T12:57:55.200998",
        "updated_at": "2021-09-13T12:57:55.201006"
    }
]
```
It is working fine but the problem is when we want below:

**1. order the data sent by `response_model`**

In the above response, `id` is returned at 2nd position but I want to return at the 1st position. Requirement for this can be:
* This is how I want my API to respond
* As a full stack developer I directly render the response in the table and my `id` column is placed in 2nd column which should be in 1st row.

**2. Update the data representation**

In the above response, we get the date as `2021-09-13T12:57:55.200998` but I want to return the formatted date which should be human-readable. This is just a single example but there can be others as explained in this [discussion](https://github.com/samuelcolvin/pydantic/discussions/3144).

### Wanted Solution

I guess we can create another magic class that lets us define the field order and value representation but that might introduce code duplication. Sorry but I don't have any perfect solution.

### Wanted Code

```python
No idea
```

### Alternatives

None. I might have to raise the same issue to pydantic.

### Operating System

Linux

### Operating System Details

Linux Mint 20.2 Cinnamon

### FastAPI Version

0.68.0

### Python Version

3.9.x

### Additional Context

None

**Comment by NoCrEaTiVe** [comment]
@jd-0001 
```
class DemoOut(BaseModel):
    id: int
    username: str
    created_at: datetime
    updated_at: datetime
    class Config:
        orm_mode = True
        validate_assignment = True

    @root_validator
    def change_created_at(cls, values):
        values["created_at"] = values["created_at"].strftime("%Y-%m-%dT%H:%M:%SZ")
        values['created_at'] = datetime.datetime.strptime(
            values["created_at"], "%Y-%m-%dT%H:%M:%SZ"
        )
        return values

    @root_validator
    def change_updated_at(cls, values):
        values["updated_at"] = values["updated_at"].strftime("%Y-%m-%dT%H:%M:%SZ")
        values['updated_at'] = datetime.datetime.strptime(
            values["updated_at"], "%Y-%m-%dT%H:%M:%SZ"
        )
        return values`
```

**Comment by jd-solanki** [comment]
Hi @NoCrEaTiVe 

I know I can use validators but this is not what they are made for. As per their name they are for validation.

**Comment by Kludex** [comment, marked as answer]
https://github.com/samuelcolvin/pydantic/issues/593

**Comment by andrewdmalone** [comment]
@jd-solanki I had this same issue and I think I've got a reasonable workaround. Make an empty subclass that inherits all of its properties from your desired superclass and a "mirror" superclass that has the additional properties that you want. Based on my testing, the properties on "actual_sub_foo" in the below example will populate in reverse order both on the FastAPI Swagger page and in any responses. See below:


```
from pydantic import BaseModel

class base_foo(BaseModel):
    prop_2: int
    prop_3: int

class mirror_sub_foo(BaseModel):
    prop_1: int

class actual_sub_foo(base_foo, mirror_sub_foo):
    None
```

In this example, if you use "actual_sub_foo" as your response_model, the API response should look like:

```
{
    "prop_1": some_int,
    "prop_2": some_int,
    "prop_3": some_int
}
```
